Added english documentation for extended TTL syntax (#8261)
ClickHouse/ClickHouse: docs/en/operations/table_engines/mergetree.md

    If a sampling expression is used, the primary key must contain it. Example: `SAMPLE BY intHash32(UserID) ORDER BY (CounterID, EventDate, intHash32(UserID))`.

> - `TTL` — A list of rules specifying durations of storage of rows and defining logic of automatic moving parts between disks and volumes.

Reviewer: ```suggestion
- `TTL` — A list of rules specifying storage duration of rows and defining logic of automatic parts movement between disks and volumes.
```
also probably needs a link to docs on how to specify disks/volumes in configuration

    Expression must have one `Date` or `DateTime` column as a result. Example:
    `TTL date + INTERVAL 1 DAY`

>     Type of the rule `DELETE|TO DISK 'xxx'|TO VOLUME 'xxx'` specified an action which is to be done with the part if expression is satisfied (reaches current time): deletion of expired rows, moving of part (if expression satisfies for all rows of a part) to specified disk (`TO DISK 'xxx'`) or volume (`TO VOLUME 'xxx'`). Default type of the rule is deletion (`DELETE`). Only one `DELETE` expression can be specified in the list of rules.

Reviewer: ```suggestion
    Type of the rule `DELETE|TO DISK 'xxx'|TO VOLUME 'xxx'` specifies an action to be done with the part if the expression is satisfied (reaches current time): removal of expired rows, moving a part (if expression is satisfied for all rows in a part) to specified disk (`TO DISK 'xxx'`) or to volume (`TO VOLUME 'xxx'`). Default type of the rule is deletion (`DELETE`). List of multiple rules can specified, but there should be no more than one `DELETE` rule.
```
probably needs a mention of what happens if all rows in a part satisfy multiple rules

Determines the lifetime of values.

The `TTL` clause can be set for the whole table and for each individual column. Table-level TTL can also specify logic of automatic move of data between disks and volumes.

Expressions must evaluate to [Date](../../data_types/date.md) or [DateTime](../../data_types/datetime.md) data type.

> To define the lifetime of data in a column, for example:

Reviewer: ```suggestion
Example:
```

> When data in a table expires, ClickHouse deletes all corresponding rows.

Reviewer: this sentence is kind of incorrect now and probably could be just removed

Author: That's right.

> For a table one can set an expression for deletion of expired rows, and expressions which if satisfied will trigger automatic move of part to [another disk or volume](#table_engine-mergetree-multiple-volumes). When rows in the table expire, ClickHouse deletes all corresponding rows.

Reviewer: ```suggestion
Table can have an expression for deletion of expired rows, and multiple expressions for automatic move of parts between [disks or volumes](#table_engine-mergetree-multiple-volumes). When rows in the table expire, ClickHouse deletes all corresponding rows. For parts moving feature, all rows of a part must satisfy the movement expression criteria.
```
